src/Uuid/Uuid.php

```diff
@@ -228,13 +228,18 @@ final public static function is(
 		string $string,
 		string|null $type = null
 	): bool {
-		if ($type !== null) {
-			return Str::startsWith($string, $type . '://');
+		$type  ??= implode('|', Uri::$schemes);
+		$pattern = sprintf('!^(%s)://([a-zA-Z0-9]*)!', $type);
```

Comment:
Custom generators could use other character sets:

```suggestion
		$pattern = sprintf('!^(%s)://(.*)!', $type);
```

Reply:
But wouldn't those clash with our caches - so we should enforce this set?

Reply:
The file cache will still work, it will just not be as pretty with the hash. Other caches should be completely unaffected.

Reply:
Didn't we just had the talk that the file cache especially doesn't work with e.g. underscores https://github.com/getkirby/kirby/blob/main/src/Cache/FileCache.php#L95-L98?

Reply:
Ah I see, what you mean with the hash.... but why then did we threw out underscores from our UUID generator?

Reply:
Because we didn't want that hash by default for the 99% use case.

Reply:
Tested it @distantnative – the original edit indeed resolves the issue when sticking to the core generator.

As for the 1% use case, it might be worth considering at least all possible ID formats and their character sets that were offerend by AutoID out of the box, as I imagine those to be the most likely cases to keep their old ID format and implement a custom generator when migrating?
- IncrementingGenerator `[0-9]`
- NanoGenerator `[A-Za-z0-9_-]{21}`
- TokenGenerator `[a-zA-Z0-9]{8}`
- UUIDGenerator `[a-f0-9-]{32}`

In other words: in favour of the suggested edit. The only way to truly solve this would be to make the regex configurable when implementing a custom generator, but that may be overkill and add unnecessary complexity. The baseline definition of a Kirby UUID is `{type}://{something}`. I keep using use my own custom format, and will simply implement my own verification method alongside my generator.